lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/test/java/io/lighty/applications/rnc/module/RncLightyModuleSmokeTest.java

```diff
@@ -10,30 +10,120 @@
 
     @Test
-    public void rncLightyModuleSmokeTest() throws ConfigurationException {
-        final RncLightyModule rncModule = new RncLightyModule(RncLightyModuleConfigUtils.loadDefaultConfig(),
-                MODULE_TIMEOUT);
-        rncModule.initModules();
-        rncModule.close();
+    public void rncLightyModuleDefaultConfigTest() throws Exception {
+        final var rncModule = new RncLightyModule(RncLightyModuleConfigUtils.loadDefaultConfig(), MODULE_TIMEOUT);
+        assertTrue(rncModule.initModules());
+        final var http1Client = getHttp1Client();
+        final var httpResponse = http1Client.send(createGetRequest(HTTP_URI + TOPOLOGY_PATH), BodyHandlers.ofString());
+        assertEquals(HttpStatus.OK_200, httpResponse.statusCode());
+        assertTrue(rncModule.close());
     }
 
+    @Test
+    public void rncLightyModuleCustomConfigTest() throws Exception {
+        final var resource = RncLightyModuleSmokeTest.class.getResource("/smokeTestConfig.json");
+        final var rncConfig = RncLightyModuleConfigUtils.loadConfigFromFile(Paths.get(resource.getPath()));
+        final var rncModule = new RncLightyModule(rncConfig, MODULE_TIMEOUT);
+        assertTrue(rncModule.initModules());
+        final var http2Client = getHttp2Client();
+        final var httpResponse = http2Client.send(createGetRequest(HTTPS_URI + TOPOLOGY_PATH), BodyHandlers.ofString());
+        assertEquals(HttpStatus.OK_200, httpResponse.statusCode());
+        assertTrue(rncModule.close());
+    }
+
+
     @Test
     public void rncLightyModuleStartFailed() throws ConfigurationException {
-        final RncLightyModuleConfiguration config = spy(RncLightyModuleConfigUtils.loadDefaultConfig());
+        final var config = spy(RncLightyModuleConfigUtils.loadDefaultConfig());
         when(config.getControllerConfig()).thenReturn(null);
-        RncLightyModule rncModule = new RncLightyModule(config, MODULE_TIMEOUT);
-        final Boolean isStarted = rncModule.initModules();
+        final var rncModule = new RncLightyModule(config, MODULE_TIMEOUT);
+        final var isStarted = rncModule.initModules();
+        final var isClose = rncModule.close();
 
         assertFalse(isStarted);
+        assertTrue(isClose);
+    }
+
+    private HttpClient getHttp2Client() throws Exception {
+        final var ssl = SSLContext.getInstance("SSL");
+        ssl.init(null, getTrustedAll(), new SecureRandom());
+        return HttpClient.newBuilder()
+                .sslContext(ssl)
+                .build();
+    }
+
+    private HttpClient getHttp1Client() {
+        return HttpClient.newHttpClient();
+    }
+
+    private HttpRequest createGetRequest(final String path) {
+        return HttpRequest.newBuilder()
+                .uri(URI.create(path))
+                .timeout(Duration.ofMinutes(1))
+                .build();
+    }
+
+    private TrustManager[] getTrustedAll() {
```

Comment:
`private static`

Isn't there anything already prepared which accepts all certificates?

Reply:
Fixed. No, at least I don't know about that.